src/openforms/forms/api/viewsets.py

```diff
@@ -388,6 +391,45 @@ def export(self, request, *args, **kwargs):
         response["Content-Length"] = len(response.content)
         return response
 
+    @extend_schema(
+        summary=_("Set default language for form"),
+        tags=["forms"],
+        request=None,
+        responses={status.HTTP_201_CREATED: FormSerializer},
+        parameters=[
+            UUID_OR_SLUG_PARAMETER,
+            OpenApiParameter(
+                name="Location",
+                type=OpenApiTypes.URI,
+                location=OpenApiParameter.HEADER,
+                description="URL of the created resource",
+                response=[status.HTTP_201_CREATED],
+            ),
+        ],
+    )
+    @transaction.atomic
+    @action(
+        detail=True,
+        methods=["post"],
+        authentication_classes=(),
+        permission_classes=(),
+    )
+    def set_default_language(self, request, *args, **kwargs):
+        """
+        Set the default language of a form.
+
+        In case translation is not enabled for the Form, the default language is set to
+        settings.LANGUAGE_CODE.
+        """
+        response = Response(status=status.HTTP_204_NO_CONTENT)
+        instance = self.get_object()
+
+        if not instance.translation_enabled:
+            translation.activate(settings.LANGUAGE_CODE)
+            response = set_language_cookie(response, settings.LANGUAGE_CODE)
+
```

Comment:
I think we want to reply to the frontend with what the currently active language is so the internal state can be updated and the appropriate frontend translations loaded, without having to make an additional API call to retrieve the current language.

Reply:
Added in https://github.com/open-formulieren/open-forms/pull/2261/commits/58950cee3e76ac98cc7d4c0238d8df673f43cdcb

By the way, do you prefer separate commits for PR feedback, or rebasing and adding changes to the existing commits?

Reply:
I don't really have a preference except for when a fix needs to be backported, then definitely rebasing.